test/replication-luatest/replicaset_name_test.lua

```diff
@@ -20,6 +20,7 @@ g.before_all = function(lg)
             server.build_listen_uri('replica', lg.replica_set.id),
         },
         replication_timeout = 0.1,
+        replication_sync_timeout = 300,
```

Comment:
Hi! Thanks for the patch!
Please consider the following changes in `replicaset_name`, `instance_name` and `cluster_name` tests:
1. `instance_name` test also fails occasionally with the patch applied, so it deserves a fix.

<details><summary>diff</summary>

```diff
diff --git a/test/replication-luatest/instance_name_test.lua b/test/replication-luatest/instance_name_test.lua
index b9f4d4a86..dc890b011 100644
--- a/test/replication-luatest/instance_name_test.lua
+++ b/test/replication-luatest/instance_name_test.lua
@@ -21,6 +21,7 @@ g.before_all = function(lg)
         },
         replication_timeout = 0.1,
         instance_name = 'master-name',
+        replication_sync_timeout = 300,
     }
     lg.master = lg.replica_set:build_and_add_server({
         alias = 'master',
```
</details>

2. Not every `box.cfg` needs a `replication_sync_timeout = 300`. `replication_sync_timeout` only affects `box.cfg{replication=...}` calls, so in `cluster_name` and `replicaset_name` tests I left it only in the initial `box.cfg` and removed all the other occurences.

<details><summary>diff</summary>

```diff
diff --git a/test/replication-luatest/cluster_name_test.lua b/test/replication-luatest/cluster_name_test.lua
index 57cd9d821..5f84d8e99 100644
--- a/test/replication-luatest/cluster_name_test.lua
+++ b/test/replication-luatest/cluster_name_test.lua
@@ -152,7 +152,6 @@ g.test_cluster_rename = function(lg)
         box.cfg{
             force_recovery = true,
             cluster_name = 'TEST2',
-            replication_sync_timeout = 300,
         }
         t.assert_equals(box.cfg.cluster_name, 'test2')
         t.assert_equals(_G.last_event.cluster_name, 'test2')
@@ -224,7 +223,6 @@ g.test_cluster_name_bootstrap_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             cluster_name = box.NULL,
-            replication_sync_timeout = 300,
         }
         t.assert_equals(box.info.cluster.name, 'test-name')
         box.space._schema:delete{'cluster_name'}
@@ -423,7 +421,6 @@ g.test_cluster_name_change_batch = function(lg)
         box.cfg{
             force_recovery = true,
             replication = {},
-            replication_sync_timeout = 300,
         }
     end)
     lg.master:exec(function()
diff --git a/test/replication-luatest/replicaset_name_test.lua b/test/replication-luatest/replicaset_name_test.lua
index 1edaeb2f4..9bad0a001 100644
--- a/test/replication-luatest/replicaset_name_test.lua
+++ b/test/replication-luatest/replicaset_name_test.lua
@@ -124,7 +124,6 @@ g.test_replicaset_rename = function(lg)
         box.cfg{
             force_recovery = false,
             replicaset_name = 'test',
-            replication_sync_timeout = 300,
         }
         t.assert_equals(_G.last_event.replicaset_name, 'test')
         t.assert_equals(box.info.replicaset.name, box.cfg.replicaset_name)
@@ -152,7 +151,6 @@ g.test_replicaset_rename = function(lg)
         box.cfg{
             force_recovery = true,
             replicaset_name = 'TEST2',
-            replication_sync_timeout = 300,
         }
         t.assert_equals(box.cfg.replicaset_name, 'test2')
         t.assert_equals(_G.last_event.replicaset_name, 'test2')
@@ -222,7 +220,6 @@ g.test_replicaset_name_bootstrap_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             replicaset_name = box.NULL,
-            replication_sync_timeout = 300,
         }
         t.assert_equals(box.info.replicaset.name, 'test-name')
         box.space._schema:delete{'replicaset_name'}
@@ -318,7 +315,6 @@ g.test_replicaset_name_recovery_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             replicaset_name = box.NULL,
-            replication_sync_timeout = 300,
         }
         box.space._schema:delete{'replicaset_name'}
     end)
@@ -377,7 +373,6 @@ g.test_replicaset_name_recovery_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             replicaset_name = box.NULL,
-            replication_sync_timeout = 300,
         }
         box.space._schema:delete{'replicaset_name'}
     end)
@@ -455,7 +450,6 @@ g.test_replicaset_name_subscribe_request_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             replicaset_name = 'test-name-new',
-            replication_sync_timeout = 300,
         }
         t.assert_equals(box.info.replicaset.name, 'test-name-new')
     end)
@@ -466,7 +460,6 @@ g.test_replicaset_name_subscribe_request_mismatch = function(lg)
         box.cfg{
             force_recovery = true,
             replication = {},
-            replication_sync_timeout = 300,
         }
         box.cfg{replication = replication}
     end, {lg.replica.box_cfg.replication})
@@ -485,7 +478,6 @@ g.test_replicaset_name_subscribe_request_mismatch = function(lg)
         box.cfg{
             force_recovery = false,
             replication = {},
-            replication_sync_timeout = 300,
         }
     end)
     lg.master:exec(function()
```
</details>

With these 2 diffs applied, the tests are 100% stable on my machine (checked with`./test-run.py $(yes _name_test | head -n 128) --exclude engine`).

Reply:
Thanks, got it. Fixed.